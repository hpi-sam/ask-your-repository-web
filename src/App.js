// @flow
import React from 'react';
import './App.scss';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <p>
          The Deployment works!!!
        </p>
      </header>
    </div>
  );
}

export default App;
