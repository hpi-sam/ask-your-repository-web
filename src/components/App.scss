@import '../style/breakpoints.scss';

.App {
  height: 100%;

  &__inner {
    max-width: var(--max-width);
    height: calc(100% - var(--navbar-height));

    margin: 0 auto;
    padding: 0 var(--padding-sides) 0px var(--padding-sides);
    padding-top: var(--navbar-height);

    display: flex;
    align-items: flex-start;
    justify-content: center;

    @media (min-width: $xs-to-sm) {
      transition: margin-left var(--sidebar-animation-duration);
      &--with-sidebar { margin-left: var(--team-sidebar-width); }
    }

    &--without-navbar {
      height: 100%;
      padding: auto;
    }
  }
}
