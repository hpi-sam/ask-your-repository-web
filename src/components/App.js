// @flow
import React from 'react';
import { connect } from 'react-redux';
import { Switch, Route, Redirect } from 'react-router';
import classNames from 'classnames';
import LoginForm from './auth/LoginForm';
import RegisterForm from './auth/RegisterForm';
import FlashMessages from './utility/flash/FlashMessages';
import ImagesIndex from './images/ImagesIndex';
import Upload from './upload/Upload';
import NavBar from './navbar/NavBar';
import Presentation from './presentation/Presentation';
import TeamSidebar from './team_sidebar/TeamSidebar';
import type { AppState } from '../state/AppState';
import ImageDetails from './images/ImageDetails';
import ImageEdit from './images/ImageEdit';
import './App.scss';

type Props = {
  isTeamSidebarOpen: boolean,
};

function App(props: Props) {
  return (
    <div className="App">
      <FlashMessages />
      <TeamSidebar />
      <NavBar />
      <div className={classNames('App__inner', { 'App__inner--with-sidebar': props.isTeamSidebarOpen })}>
        <Switch>
          <Redirect exact from="/" to="/images" />
          <Route path="/upload" component={Upload} />
          <Route path="/images/:id/edit" component={ImageEdit} />
          <Route path="/images/:id" component={ImageDetails} />
          <Route path="/images" component={ImagesIndex} />
          <Route path="/presentation" component={Presentation} />
          <Route path="/login" component={LoginForm} />
          <Route path="/register" component={RegisterForm} />
        </Switch>
      </div>
    </div>
  );
}

const mapStateToProps = (state: AppState) => ({
  isTeamSidebarOpen: state.teamSidebar.isOpen,
});

export default connect(mapStateToProps)(App);
