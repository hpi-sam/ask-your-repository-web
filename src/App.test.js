// @flow
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

describe('<App />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
