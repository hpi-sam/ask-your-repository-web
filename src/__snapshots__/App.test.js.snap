// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <p>
      HP2018HG1 - Website
    </p>
  </header>
</div>
`;
